core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -1008,7 +1010,23 @@ public void syncPersistDirectory(InodeDirectory dir, JournalContext journalConte
           MkdirsOptions mkdirsOptions =
               MkdirsOptions.defaults().setCreateParent(false).setOwner(dir.getOwner())
                   .setGroup(dir.getGroup()).setMode(new Mode(dir.getMode()));
-          ufs.mkdirs(ufsUri, mkdirsOptions);
+          if (!ufs.mkdirs(ufsUri, mkdirsOptions)) {
```

Comment:
Does making this change here fix the load metadata case? Why does load metadata involve persisting directories to the UFS?

Reply:
It works because currently both load metadata and create directory use `INodeTree.createPath` to create the Inodes leading to the target node. For INodes in the path that only exist in UFS we just create the INodes using the owner and modes passed in which should only apply for the target INode and new directories which are created. Given that we call `syncPersistDirectory` on each directory created, we can determine if there is already UFS directory exists and apply directory status to the new Inode.

Reply:
Well, one interesting scenario is trying to load metadata, but during the process, someone deletes the underlying directories. In that scenario, what is the correct behavior? Should we just fail the "create path"? The current PR will try to mkdir the UFS dir again.

These types of races are always possible, since we do not control the UFS modifications. We will just have to choose a behavior.